daemon/cluster/convert/swarm.go

```diff
@@ -0,0 +1,112 @@
+package convert
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/crypto/bcrypt"
+
+	types "github.com/docker/engine-api/types/swarm"
+	swarmapi "github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/protobuf/ptypes"
+)
+
+// SwarmFromGRPC converts a grpc Cluster to a Swarm.
+func SwarmFromGRPC(c swarmapi.Cluster) types.Swarm {
+	swarm := types.Swarm{
+		ID: c.ID,
+		Spec: types.Spec{
+			Orchestration: types.OrchestrationConfig{
+				TaskHistoryRetentionLimit: c.Spec.Orchestration.TaskHistoryRetentionLimit,
+			},
+			Raft: types.RaftConfig{
+				SnapshotInterval:           c.Spec.Raft.SnapshotInterval,
+				KeepOldSnapshots:           c.Spec.Raft.KeepOldSnapshots,
+				LogEntriesForSlowFollowers: c.Spec.Raft.LogEntriesForSlowFollowers,
+				HeartbeatTick:              c.Spec.Raft.HeartbeatTick,
+				ElectionTick:               c.Spec.Raft.ElectionTick,
+			},
+			Dispatcher: types.DispatcherConfig{
+				HeartbeatPeriod: c.Spec.Dispatcher.HeartbeatPeriod,
+			},
+		},
+	}
+
+	swarm.Spec.CAConfig.NodeCertExpiry, _ = ptypes.Duration(c.Spec.CAConfig.NodeCertExpiry)
+
+	// Meta
+	swarm.Version.Index = c.Meta.Version.Index
+	swarm.CreatedAt, _ = ptypes.Timestamp(c.Meta.CreatedAt)
+	swarm.UpdatedAt, _ = ptypes.Timestamp(c.Meta.UpdatedAt)
+
+	// Annotations
+	swarm.Spec.Name = c.Spec.Annotations.Name
+	swarm.Spec.Labels = c.Spec.Annotations.Labels
+
+	for _, policy := range c.Spec.AcceptancePolicy.Policies {
+		swarm.Spec.AcceptancePolicy.Policies = append(swarm.Spec.AcceptancePolicy.Policies, types.Policy{
+			Role:       types.NodeRole(policy.Role.String()),
+			Autoaccept: policy.Autoaccept,
+			Secret:     string(policy.Secret.Data),
```

Comment:
Panicked here when running `docker swarm inspect`.
```
2016-06-12 15:02:57.184744 I | http: panic serving @: runtime error: invalid memory address or nil pointer dereference
goroutine 234 [running]:
net/http.(*conn).serve.func1(0xc8203c5b80)
        /usr/local/go/src/net/http/server.go:1389 +0xc1
panic(0x1a6c400, 0xc820010070)
        /usr/local/go/src/runtime/panic.go:443 +0x4e9
github.com/docker/docker/daemon/cluster/convert.SwarmFromGRPC(0xc820ee0280, 0x19, 0x9, 0xc8208c5e40, 0xc8208c5e50, 0xc8208c5e11, 0x7, 0x0, 0xc8208c5ea0, 0x2, ...)
        /go/src/github.com/docker/docker/daemon/cluster/convert/swarm.go:50 +0x267
github.com/docker/docker/daemon/cluster.(*Cluster).Inspect(0xc8200b9680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/docker/docker/daemon/cluster/cluster.go:411 +0x23f
github.com/docker/docker/api/server/router/swarm.(*swarmRouter).inspectCluster(0xc82072a7e0, 0x7f350b7d37d8, 0xc82089db30, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0, 0xc82089d6e0, 0x0, 0x0)
        /go/src/github.com/docker/docker/api/server/router/swarm/cluster_routes.go:48 +0x67
github.com/docker/docker/api/server/router/swarm.(*swarmRouter).(github.com/docker/docker/api/server/router/swarm.inspectCluster)-fm(0x7f350b7d37d8, 0xc82089db30, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0, 0xc82089d6e0, 0x0, 0x0)
        /go/src/github.com/docker/docker/api/server/router/swarm/cluster.go:30 +0x74
github.com/docker/docker/api/server/middleware.VersionMiddleware.WrapHandler.func1(0x7f350b7d37d8, 0xc82089db30, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0, 0xc82089d6e0, 0x0, 0x0)
        /go/src/github.com/docker/docker/api/server/middleware/version.go:56 +0x9aa
github.com/docker/docker/api/server/middleware.UserAgentMiddleware.WrapHandler.func1(0x7f350b7d37d8, 0xc82089d980, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0, 0xc82089d6e0, 0x0, 0x0)
        /go/src/github.com/docker/docker/api/server/middleware/user_agent.go:45 +0x68e
github.com/docker/docker/api/server/middleware.DebugRequestMiddleware.func1(0x7f350b7c5940, 0xc8200110f8, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0, 0xc82089d6e0, 0x0, 0x0)
        /go/src/github.com/docker/docker/api/server/middleware/debug.go:21 +0xd4d
github.com/docker/docker/api/server.(*Server).makeHTTPHandler.func1(0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0)
        /go/src/github.com/docker/docker/api/server/server.go:139 +0x138
net/http.HandlerFunc.ServeHTTP(0xc820859820, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0)
        /usr/local/go/src/net/http/server.go:1618 +0x3a
github.com/gorilla/mux.(*Router).ServeHTTP(0xc82000d0e0, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0)
        /go/src/github.com/docker/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x29e
github.com/docker/docker/api/server.(*routerSwapper).ServeHTTP(0xc8206c71a0, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0)
        /go/src/github.com/docker/docker/api/server/router_swapper.go:29 +0x72
net/http.serverHandler.ServeHTTP(0xc8202a7f80, 0x7f3508607e78, 0xc8203fa680, 0xc8214a5dc0)
        /usr/local/go/src/net/http/server.go:2081 +0x19e
net/http.(*conn).serve(0xc8203c5b80)
        /usr/local/go/src/net/http/server.go:1472 +0xf2e
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2137 +0x44e
```

Steps to reproduce:
1. Start docker service with `/usr/bin/dockerd --debug --raw-logs`
2. `docker swarm init`
3. `docker swarm inspect`